src/particleContainer/AutoPasContainer.cpp

```diff
@@ -291,26 +291,67 @@ void AutoPasContainer::traverseTemplateHelper() {
+		allSame &= _particlePropertiesLibrary.getSigmaSquare(i) == sigmasqFirstComponent;
+	}
+	bool useMixing = not allSame;
+
+	if (useMixing) {
+		global_log->debug() << "AutoPasContainer: Using mixing." << std::endl;
+		if (_useAVXFunctor) {
+			// Generate the functor. Should be regenerated every iteration to wipe internally saved globals.
+			autopas::LJFunctorAVX<Molecule, /*applyShift*/ shifting, /*mixing*/ true, autopas::FunctorN3Modes::Both,
+								  /*calculateGlobals*/ true>
+				functor(_cutoff, _particlePropertiesLibrary);
+
+			// here we call the actual autopas' iteratePairwise method to compute the forces.
+			_autopasContainer.iteratePairwise(&functor);
+			upot = functor.getUpot();
+			virial = functor.getVirial();
+		} else {
+			// Generate the functor. Should be regenerated every iteration to wipe internally saved globals.
+			autopas::LJFunctor<Molecule, /*applyShift*/ shifting, /*mixing*/ true, autopas::FunctorN3Modes::Both,
+							   /*calculateGlobals*/ true>
+				functor(_cutoff, _particlePropertiesLibrary);
+
+			// here we call the actual autopas' iteratePairwise method to compute the forces.
+			_autopasContainer.iteratePairwise(&functor);
+			upot = functor.getUpot();
+			virial = functor.getVirial();
+		}
 	} else {
-		// Generate the functor. Should be regenerated every iteration to wipe internally saved globals.
-		autopas::LJFunctor<Molecule, /*applyShift*/ shifting, /*mixing*/ true, autopas::FunctorN3Modes::Both,
-						   /*calculateGlobals*/ true>
-			functor(_cutoff, _particlePropertiesLibrary);
-
-		// here we call the actual autopas' iteratePairwise method to compute the forces.
-		_autopasContainer.iteratePairwise(&functor);
-		upot = functor.getUpot();
-		virial = functor.getVirial();
+		global_log->debug() << "AutoPasContainer: Not using mixing." << std::endl;
+		if (_useAVXFunctor) {
+			// Generate the functor. Should be regenerated every iteration to wipe internally saved globals.
+			autopas::LJFunctorAVX<Molecule, /*applyShift*/ shifting, /*mixing*/ false, autopas::FunctorN3Modes::Both,
+								  /*calculateGlobals*/ true>
+				functor(_cutoff);
+			functor.setParticleProperties(epsilon24FirstComponent, sigmasqFirstComponent);
+
+			// here we call the actual autopas' iteratePairwise method to compute the forces.
+			_autopasContainer.iteratePairwise(&functor);
+			upot = functor.getUpot();
+			virial = functor.getVirial();
+		} else {
+			// Generate the functor. Should be regenerated every iteration to wipe internally saved globals.
+			autopas::LJFunctor<Molecule, /*applyShift*/ shifting, /*mixing*/ false, autopas::FunctorN3Modes::Both,
+							   /*calculateGlobals*/ true>
+				functor(_cutoff);
+			functor.setParticleProperties(epsilon24FirstComponent, sigmasqFirstComponent);
+
+			// here we call the actual autopas' iteratePairwise method to compute the forces.
+			_autopasContainer.iteratePairwise(&functor);
+			upot = functor.getUpot();
+			virial = functor.getVirial();
```

Comment:
This is at the border of being enough code duplication to justify a template helper function, which takes the functor as a template argument.

Reply:
done.